app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -58,22 +58,17 @@
 
     private final static String TAG = "SmsDetector";
 
-    private DataInputStream dis;
     private DataOutputStream dos;
+    BufferedReader in;
```

Comment:
Is there any reason why this field isn't encapsulated? It's a buffer and its scope should be minimal to avoid memory leaks or running out of memory.